src/executors/executor_sycl.hpp

```diff
@@ -26,6 +26,8 @@
 #ifndef SYCL_BLAS_EXECUTOR_SYCL_HPP
 #define SYCL_BLAS_EXECUTOR_SYCL_HPP
 
+#include <algorithm>
```

Comment:
Is it worth including the algorithm header just for the single usage of std::min? I was under the impression that it is quite heavyweight.

Reply:
You're right that it might affect a bit the compilation time, I fix that in the reduction PR since this code belongs there (this one is only the benchmark, as said in the description the changes from #167 are only here for convenience and will disappear once the other is merged).